### PR TITLE
fix(components/ag-grid): update data grid styles

### PR DESCRIPTION
**Pull request by johnhwhite**
[AB#2392242](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2392242)

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/1177/storybook/)

Component Storybooks:

- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/1177/storybooks/ag-grid-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/1177/storybooks/layout-storybook/)
- [modals-storybook](https://blackbaud.github.io/skyux-pr-preview/1177/storybooks/modals-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/1177/storybooks/pages-storybook/)
- [split-view-storybook](https://blackbaud.github.io/skyux-pr-preview/1177/storybooks/split-view-storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/1177/storybooks/tabs-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/1177/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/1177/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/1177/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/1177/playground/)

**Comment by Blackbaud-TrevorBurch** [comment]
When I view the "Edit complex cells" page - I see a couple things

1. The "Validation autocomplete" header is cut off
2.The "Validation autocomplete" header includes what appears to be a built-in AG Grid filter button

**Comment by Blackbaud-TrevorBurch** [comment]
I could go down to a 2 on risk level here. But putting 3 because it does change AG Grid modern stylings a lot

File: libs/components/ag-grid/src/lib/styles/_base.scss [hunk 1]
```diff
@@ -7,6 +7,7 @@
     suppress-native-widget-styling: true,
     themes: (
       sky-default: (
+        --ag-icon-font-family: 'agGridAlpine',
```
Comment (johnhwhite) [inline libs/components/ag-grid/src/lib/styles/_base.scss]: This setting provides a fallback set of icons where we have not overridden the icon.

File: libs/components/data-manager/src/lib/modules/data-manager/data-manager-toolbar/data-manager-toolbar.component.html [hunk 2]
```diff
@@ -59,16 +59,15 @@
         <sky-radio-group
           *ngIf="activeView && views && views.length > 1"
           [ariaLabel]="'data view switcher'"
+          [(ngModel)]="activeView.id"
```
Comment (Blackbaud-ErikaMcVey) [inline libs/components/data-manager/src/lib/modules/data-manager/data-manager-toolbar/data-manager-toolbar.component.html]: thanks for fixing this

File: libs/components/flyout/src/lib/modules/flyout/flyout.component.scss [hunk 3]
```diff
@@ -150,6 +151,7 @@ $header-height: 50px;
 @include mixins.sky-theme-modern-dark {
   .sky-flyout {
     background: #000;
+    --sky-background-color-page-default: #000;
```
Comment (Blackbaud-ErikaMcVey) [inline libs/components/flyout/src/lib/modules/flyout/flyout.component.scss:154]: why do flyout and modal need to set the background color variable now?

File: libs/components/ag-grid/src/lib/modules/ag-grid/ag-grid.service.ts [hunk 4]
```diff
@@ -114,6 +114,81 @@ function getValidatorCellRendererSelector(component: string, fallback?: any) {
...
+  previous: {
+    faIcon: 'caret-left',
+    skyIcon: 'chevron-left',
+  },
+  first: {
+    faIcon: 'step-backward',
+    skyIcon: 'double-chevron-left',
+  },
+};
+type iconMapType = typeof iconMap;
```
Comment (Blackbaud-ErikaMcVey) [inline libs/components/ag-grid/src/lib/modules/ag-grid/ag-grid.service.ts]: I think this type should probably be capitalized `IconMapType`. It also might not hurt to move these into another file

File: libs/components/flyout/src/lib/modules/flyout/flyout.component.scss (hunk 3 above)
Comment (johnhwhite) [reply libs/components/flyout/src/lib/modules/flyout/flyout.component.scss:154]: The designer wants `transparent` background, but when viewkeeper pins the header, `transparent` makes the scrolled content viewable under the header. I'm relying on `--sky-background-color-page-default` and updating it in contexts where the background changes, which is flyout and modal.

File: libs/components/ag-grid/src/lib/modules/ag-grid/ag-grid.service.ts (hunk 4 above)
Comment (johnhwhite) [reply libs/components/ag-grid/src/lib/modules/ag-grid/ag-grid.service.ts]: Good call. Moved to a separate file.

**Review by Blackbaud-ErikaMcVey — DISMISSED** [review]

**Review by Blackbaud-ErikaMcVey — APPROVED** [review]

**Pull request merged**